io/zenoh-link-commons/src/listener.rs

```diff
@@ -109,12 +111,17 @@ impl ListenersUnicastIP {
         let guard = zread!(self.listeners);
         for (key, value) in guard.iter() {
             let (kip, kpt) = (key.ip(), key.port());
+            let iface = value
+                .endpoint
+                .config()
+                .get(BIND_INTERFACE)
+                .map(|s| s.to_string());
```

Comment:
I think it is possible to avoid the `map(|s| to_string())` here if `get_local_addresses(..)` takes `Option<&str>`.